plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/CacheManager.java

```diff
@@ -30,4 +44,62 @@ void createCache() {
             }
         }
     }
+
+    public void addToCache(String cacheKey, String value) {
+        Path fileToCache = cache.resolve(cacheKey + ".json");
+        try (Writer writer = Files.newBufferedWriter(fileToCache, StandardCharsets.UTF_8)) {
+            writer.write(value);
+        } catch (IOException e) {
+            throw new ModernizerException("Unable to add cache", e);
+        }
+    }
+
+    /**
+     * Retrieves a json object from the cache.
+     * <p>
+     * Will return null if the key can't be found or if it hasn't been
+     * modified for 1 hour
+     *
+     * @param cacheKey key to lookup, i.e. update-center
+     * @return the cached json object as a string or null
+     */
+    public String retrieveFromCache(String cacheKey) {
```

Comment:
I would introduce a simple POJO

```suggestion
    public CacheItem retrieveFromCache(String cacheKey) {
``

Reply:
Also a bit more context. If we add a class CacheItem (abstract or not) it means that we can cache entities like `Plugin` or other content